pkg/cloudprovider/vsphereparavirtual/cloud.go

```diff
@@ -166,8 +166,11 @@ func (cp *VSphereParavirtual) Routes() (cloudprovider.Routes, bool) {
 }
 
 // ProviderName returns the cloud provider ID.
+// Note: Returns 'vsphere' instead of 'vsphere-paravirtual'
+// since CAPV expects the ProviderID to be in form 'vsphere://***'
+// https://github.com/kubernetes/cloud-provider-vsphere/issues/447
 func (cp *VSphereParavirtual) ProviderName() string {
-	return ProviderName
+	return "vsphere"
```

Comment:
Actually... I think I change my mind, sorry.

Let's use a constant here actually, and have two constants for each package `ProviderName` and `RegisteredProviderName`. I have concerns that the ProviderName constant is used in other places and having it set to `vsphere-paravirtual` can lead to unexpected behaviors later down the road.

Reply:
Yeah it might be used in the future in other place. People might step into similar issue

Reply:
But I checked cloud-provider `ProviderName ` was not touched, usually they will refer to `ProviderName()` func

Reply:
Anyway, I change it back

Reply:
And I ran the test again, it passed

Reply:
@andrewsykim please take a look. Thank you